crates/cxx-qt-gen/src/naming/cpp.rs

```diff
@@ -205,15 +205,18 @@ fn path_argument_to_string(
 
 /// Convert a segment of a path to C++
 fn path_segment_to_string(segment: &PathSegment, type_names: &TypeNames) -> Result<String> {
-    let mut ident = segment.ident.to_string();
+    let ident = segment.ident.to_string();
 
     // If we are a Pin<T> then for C++ it becomes just T
-    let args = match ident.as_str() {
+    let args = match &*segment.ident.to_string() {
```

Comment:
why not `&*ident` ?

Reply:
Ah, right. I considered removing the ident variable, as I don't think the variable name is quite appropriate.
But I ended up in this weird in-between state :see_no_evil: 

Fixing it.

Reply:
Done